storage/innobase/log/log0recv.cc

```diff
@@ -1138,7 +1138,9 @@ static
       return;
     }
 
-    if (!file.contains(checkpoint_in_file.m_checkpoint_lsn)) {
+    if (IF_XB(log.m_files_ctx.m_files_ruleset >
+              Log_files_ruleset::PRE_8_0_30) &&
+        !file.contains(checkpoint_in_file.m_checkpoint_lsn)) {
```

Comment:
why isn't applicable for pre 8030? checkpoint_lsn would be in the file detected above, isn't it?

Reply:
on pre-8.0.30 we always have checkpoint marker on ib_logfile0, even if the LSN it points to is located on ib_logfile1. This causes the backup to fail in case we are currently writing on ib_logfile1.

Reply:
can you please add this as a comment? Maybe for versions < 8_0_30 we can check that the front file will have checkpoint header?

Reply:
Another doubt I have is how will we find wrap around logic? because we always have two files loaded in loop. 
Unlike 8.0.30, the loaded files start_lsn will keep changing. In 8.0.30, the files start_lsn is fixed and new files appear.

Reply:
@satya-bodapati logic explained as part of commit message (adjusted)
wrap around logic has been fixed at 8d8feb3